### PR TITLE
cellMouseGetDataList fixes

### PR DESCRIPTION
**Pull request by hcorion**
Moves Dark Souls 2 from Nothing -> Ingame.

### CODE REVIEW

File: rpcs3/Emu/Cell/Modules/cellMouse.cpp [hunk 1]
```diff
@@ -134,22 +134,27 @@ s32 cellMouseGetData(u32 port_no, vm::ptr<CellMouseData> data)
...
 {
-	UNIMPLEMENTED_FUNC(sys_io);
-	
-	return CELL_OK;
+	sys_io.todo("cellMouseGetDataList(port_no=%d, data=0x%x", port_no, data);
+	if (g_cfg.io.mouse == mouse_handler::null)
+		return CELL_MOUSE_ERROR_NO_DEVICE;
+	else
+	{
+		sys_io.error("cellMouseGetDataList was called while mouse handler is not set to null, this is gonna hurt...");
```
Comment (jjsat) [inline rpcs3/Emu/Cell/Modules/cellMouse.cpp]: If data->list_num is set to 0 before returning this should hurt much less.

Comment (kd-11) [reply rpcs3/Emu/Cell/Modules/cellMouse.cpp]: Confirmed. Setting list_num to 0 fixes dark souls 2 as well

Comment (hcorion) [reply rpcs3/Emu/Cell/Modules/cellMouse.cpp]: Good idea. Fixed.

**Comment by Aske1836** [comment]
image
Thanks a lot for this! The development speed of this emulator is increasing rapidly, i love it!

**Pull request merged**